code/Model/SiteTree.php

```diff
@@ -951,16 +946,26 @@ public function setParent($item)
 
     /**
      * Get the parent of this page.
+     * This is here to keep compatability with the has_one magic method
+     * that always returns an object, even if there is none for the specific relation.
      *
-     * @return SiteTree Parent of this page
+     * @return SiteTree Parent of this page or SiteTree singleton
+     */
+    public function Parent()
+    {
+        return $this->getParent() ?: Injector::inst()->get(self::class);
```

Comment:
So calling `$RootPage->Parent` would previously return a `SiteTree` singleton?  This seems a bit odd.  Was the previously the reason for all those `$parent->exists()` checks?

Reply:
yes, because a has_one returns a singleton if there is no relationship.